### PR TITLE
[#1202] Add Javadoc for backend functions

### PR DESCRIPTION
**Pull request by HCY123902**
#1202 
Commit  message:
```
Add Javadoc to methods in StreamGobbler, CustomLogFormatter, 
FileType, and RepoConfiguration.

Some methods in these classes do not seem to have Javadoc 
although the methods are not standard getter or setter.

To enhance readability, it may be necessary to explain what 
they do, especially for run() in StreamGobbler since the method
name may not give much information.

Additionally, there seems to be a colon missing in the developer guide
architecture section.

Let's add Javadoc to these methods and add the missing punctuation
to the developer guide to make documentation more consistent.
```
The missing colon in the developer guide architecture section is something that I happen to notice when going through it. 
I previously plan to split this into 2 separate pull requests, but since only minor changes are made, I decide to combine them into one.

### CODE REVIEW

**Comment by jamessspanggg** [comment]
@HCY123902 please follow the commit message guidelines located [here](https://se-education.org/guides/guidelines/PRs.html). In particular, wrap the commit message body within two ```.

**Comment by fzdy1914** [comment]
>  In particular, wrap the commit message body within two ```.

@jamessspanggg Actually, it is OK not to wrap the commit message body with two ```, as long as they are easily distinguishable. Now, we use "`" for variable highlighting.

**Comment by HCY123902** [comment]
Thanks for the reminder. I will change it according to the guideline.

File: src/main/java/reposense/system/StreamGobbler.java [hunk 1]
```diff
@@ -24,6 +24,11 @@ public String getValue() {
         return value;
     }
 
+    /**
+     * Reads from input stream {@code is} and stores it into {@code value}.
+     * This method will attempt to iteratively read 8KB of data from {@code is}, until no
```
Comment (jamessspanggg) [inline src/main/java/reposense/system/StreamGobbler.java:29]: I think its not needed to state the second point `This method will attempt...`

Comment (HCY123902) [reply src/main/java/reposense/system/StreamGobbler.java]: Sure, I have removed it. It does seem to be a bit redundant.

**Review by jamessspanggg — APPROVED** [review]

**Pull request merged**